### PR TITLE
In channels with a discord bridge, turns "<bridge> <Sender> Message" …

### PR DESCRIPTION
**Pull request by snan**
…into "&lt;Sender> Message", and hides spoilers.

### CODE REVIEW

**Comment by vague666** [comment]
Irssi can get confused by you changing the nick like that
A better way would be to rewrite the raw message. The following has been written for trigger.pl
`Soliton ┃ xacktm: this is exactly what i'm using, perhaps easier to adapt from that: 
/trigger add -rawin -regexp ':discordbot[^!]*!~discordbo@baldras.wesnoth.org PRIVMSG #(wesnoth.*) :<([^>]+)> (.*)' -replace ':d|$2!~Soliton@wesnoth/developer/soliton PRIVMSG #$1 :$3'`

**Comment by snan** [comment]
Great idea!♥  
Pushed.

**Comment by snan** [comment]
Hold on, with that implementation it breaks when people on Discord have spaces in their display names.

**Comment by soliton-** [comment]
You can tweak the regex to allow spaces in nicks but I've never tested that. I've read what vague said as well that irssi can get confused if you force feed it invalid nicks. Might be worthwhile to just try and see what really happens though.

File: scripts/discord_unbridge.pl [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+#
+# Based on discord_unhilight by Christoffer Holmberg, in turn
+# based on slack_strip_auto_cc.pl by Ævar Arnfjörð Bjarmason.
+
+my $bridgename = "Yoda50";
+
+sub bot_nick_change {
+    my($server, $message) = @_;
+    $message =~ s/:$bridgename!~[^ ]* PRIVMSG ([&#][^ ]+) :<([^>]+)> (.*)/:$2!~Yoda50\@ToscheStation PRIVMSG $1 :$3/;
+    $message =~ s/\|\|([^|][^|]*)\|\|/1,1$1/;
```
Comment (soliton-) [inline scripts/discord_unbridge.pl:35]: I'm guessing the `1,1` changes the color. I think you may want to reset the color at the end and use the g flag to replace multiple spoilers in one message.

You can also simplify `([^|][^|]*)` to `([^|]+)`.

**Comment by snan** [comment]
The regex in that `'server incoming'` version I pushed a few versions ago already allows the space in the nick in signal it continues. The problem is further down the stream from that.

File: scripts/discord_unbridge.pl (hunk 1 above)
Comment (snan) [reply scripts/discord_unbridge.pl:35]: I already do reset the color. The g and + are good ideas, thank you for those.

**Comment by snan** [comment]
`<([^>]+)>` This is what allows spaces in Discord display names, both of our versions have that. One of my users find messages from these "space-heads" in their server log.

**Comment by soliton-** [comment]
To clarify what I'm personally using now with trigger.pl to match the nick is ``<([][[:alpha:]_{}^`|][][[:alnum:]_{}^`|\-]*)>``. But as mentioned I have no idea what bad things actually do happen with invalid IRC nicks.

**Comment by vague666** [comment]
spaces in a nick could segfault irssi but it's not something I've seen, mainly because I don't use discord nor discord with irssi

**Comment by snan** [comment]
I hate Discord.

**Comment by ailin-nemui** [comment]
maybe replace spaces, @ sign and exclamation mark with underscores if you want to cram them in the server incoming?

**Comment by soliton-** [comment]
The issue with replacing characters is that you may get false duplicates then which could be confusing. The chance of that might be so low that it can be considered acceptable though. Also I'm not sure if you can get confusing duplicates anyway when people are allowed to change their nick on the discord side...

Another issue is not knowing how to highlight (`@nick`) discord users then but I believe highlighting discord users with spaces in their nick at least does not work anyway.

Given that there probably is no perfect solution for all these issues a solution that works well enough most of the time is what you should aim for and document the shortcomings.

**Comment by snan** [comment]
I'm not sure what the shortcomings are, with the 'event privmsg'
approach that I original took, and then later reverted to when the
'server incoming' version broke for my users. Not saying there aren't
any, just that I haven't found them.

Ideally, I would want it to work on the display level, enough so you
could highlight, ignore etc the Discord users separately (instead of the
bridge as a whole), but not get confused into trying to /query them.

**Comment by vague666** [comment]
Maybe @ailin-nemui can correct me if I'm wrong when I said irssi can get confused when you changed the nick the way you did in the 'event privmsg' handler

**Comment by ailin-nemui** [comment]
@snan are you happy with the script for now?

**Comment by snan** [comment]
Yes, I haven't experienced any problems so far, though I haven't
deliberately tried to stress test it with weird situations.

**Pull request merged**